### DISCUSSION TITLE
Request of supporting on mac

### DISCUSSION THREAD

**Discussion by Jarvis-depot** [hpcaitech/ColossalAI#3394] [Community | Q&A]
Hi Sir,
When I tried "docker build -t colossalai ./docker", it told me
```
#7 15.98     No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda'
#7 15.98     False
#7 15.98     [extension]  loaded builders for cpu_adam, fused_optim, moe, multi_head_attn, scaled_masked_softmax, scaled_upper_triangle_masked_softmax, layernorm
```

Seems that only support laptop with NVIDIA CUDA
Can u add mps support to a macbook with MPS enabled ?(https://www.iotword.com/9206.html)

**Comment by binmakeswell** [comment]
Hi @Jarvis-depot Unfortunately, we currently mainly support NVIDIA GPU and Linux, which are the most mainstream development environments available.
https://github.com/hpcaitech/ColossalAI#installation
Support for other hardware and OS will be gradually developed in the future.
